### DISCUSSION TITLE
Is it Possible to linkCredential with Apple Sign-In when using JS SDK?

### DISCUSSION THREAD

**Discussion by howlettga** [capawesome-team/capacitor-firebase#712] [Q&A] [answered]
I am currently signing a user in via Apple. Based on this [answer](https://github.com/robingenz/capacitor-firebase-authentication/issues/41#issuecomment-884106449) I have to configure `skipNativeAuth: true` to get the nonce to sign in to the JavaScript library. I am using the JS layer for route protection and other in my application. Sign-in code:
```js
  async signInWithApple() {
    const result = await FirebaseAuthentication.signInWithApple({
      scopes: ['name', 'email'],
      skipNativeAuth: true,   // doesn't stay logged in to link later
    });

    const provider = new OAuthProvider('apple.com');
    provider.addScope('name');
    provider.addScope('email');
    const credential = provider.credential({
      idToken: result.credential?.idToken,
      rawNonce: result.credential?.nonce,
    });

    const userCredential = await signInWithCredential(this.auth, credential);
    return userCredential;
  }
```

I then later want to be able to link a Facebook login to the user; mostly to pull shared friends. When I use the linkWithFacebook functionality after signing in via Apple, there is no user on the native layer and so I get a `No user is signed in` error. Makes sense to me. **I'm just wondering if there is a workaround to getting these two working together while also signing into the js layer?** Can I propagate the logged in user back to the native layer once I've logged in to to the js layer? Or log into the js layer without skipping native auth (I don't see any access token or other means of instantiating the credential? Or maybe I just have to strip out uses of the firebase js library in my application to get this working...

Here's the facebook link method:
```js
  async linkWithFacebook() {
    console.log(await FirebaseAuthentication.getCurrentUser());
    const result = await FirebaseAuthentication.linkWithFacebook({
      scopes: [ 'public_profile', 'user_friends' ],
    });
    return result;
  }
```

**Comment by robingenz** [comment, marked as answer]
No, unfortunately this is not possible. If you want to use the `linkWithFacebook()` method, you have to sign in the user on the native layer. The configuration option `skipNativeAuth` was originally introduced because there were no Capacitor plugins for the other Firebase SDKs. In the meantime, however, there is a Capacitor plugin for almost every Firebase SDK so that the use of `skipNativeAuth` should only be necessary in rare cases.
